core/src/main/java/org/opensearch/sql/planner/optimizer/LogicalPlanOptimizer.java

```diff
@@ -60,6 +62,30 @@ public static LogicalPlanOptimizer create() {
         new CreateTableWriteBuilder()));
   }
 
+  /**
+   * Create {@link LogicalPlanOptimizer} with pre-defined rules.
+   */
+  public static LogicalPlanOptimizer paginationCreate() {
```

Comment:
difference is CreatePagingTableScanBuilder()?

Reply:
`PushPageSize` and `CreatePagingTableScanBuilder` instead of `CreateTableScanBuilder`